src/stub/src/powerpc64le-linux.elf-main.c

```diff
@@ -248,18 +251,23 @@ do_xmap(
         Extent xo;
         size_t mlen = xo.size = phdr->p_filesz;
         char  *addr = xo.buf  =                 (char *)phdr->p_vaddr;
-        char *haddr =           phdr->p_memsz +                  addr;
         size_t frag  = (long)addr &~ PAGE_MASK;
         mlen += frag;
-        addr -= frag;
-        addr  += reloc;
-        haddr += reloc;
+        addr  += reloc - frag;
 
         if (addr != mmap(addr, mlen, prot | (xi ? PROT_WRITE : 0),
                 MAP_FIXED | MAP_PRIVATE | (xi ? MAP_ANONYMOUS : 0),
                 (xi ? -1 : fdi), phdr->p_offset - frag) ) {
             err_exit(8);
         }
+
+        char *haddr =           phdr->p_memsz + frag +           addr;
+
+        if ( xo.buf != addr  + frag ) { // pie relocatable addr ?
```

Comment:
So there is something wrong somewhere since - when compiled with -fpie - I get add = 0 when setting haddr initally. But as reloc get the offset, then addr before mmap is good.... This is why in you do char  *addr = xo.buf  =         reloc + (char *)phdr->p_vaddr; in stub/src/amd64-linux.elf-main.c: - Agree to use a similar coding for both - Will rework it.